pkg/scheduler/framework/framework.go

```diff
@@ -701,6 +701,39 @@ func (f *framework) runSchedulingCycleForPickNPlacementType(
 			return ctrl.Result{}, err
 		}
 
+		// Check if the scheduler needs to take action; a scheduling cycle is only needed if
+		// * the policy is of the PickAll type; or
+		// * the policy is of the PickN type, and currently there are not enough number of bindings.
+		if !shouldSchedule(policy, numOfClusters, len(bound)+len(scheduled)) {
```

Comment:
should we skip the following steps if we have already done the downscale()?

should be moved to line 740?

Reply:
Yeah, I must have copied the code in the wrong place when organizing them. Fixed.